includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -470,4 +471,12 @@ protected function handle_order_propagation( $rel_order, $charge_permission_id,
 			wc_apa()->get_gateway()->add_status_change_note( $rel_order, $charge_permission_id, $new_status );
 		}
 	}
+
+	public function maybe_change_session_key( $session_key ) {
+		if ( isset( $_POST['_wcsnonce'] ) && isset( $_POST['woocommerce_change_payment'] ) ) {
```

Comment:
Because the validity will be checked elsewhere. This just changes the key used to store in the session, not something that needs securing, since the session is only valid for the user.